packages/headers/src/main.ts

```diff
@@ -45,7 +47,7 @@ export class HeadersHandler {
     return this.#headersFiles
   }
 
-  async handle(request: Request, response: Response) {
+  async apply(request: Request, response?: Response, collector?: HeadersCollector) {
```

Comment:
Instead of a `handle` method that returns (and mutates) the response with the headers applied, I'm proposing an `apply` method that can mutate a response and return the headers that were added. Additionally, an optional `headersCollector` function acts as a callback that is called with each key/value pair.